lib/streamlit/app_session.py

```diff
@@ -252,8 +252,33 @@ def request_rerun(self, client_state: Optional[ClientState]) -> None:
     def session_state(self) -> "SessionState":
         return self._session_state
 
-    def _on_source_file_changed(self) -> None:
+    def _should_rerun_on_file_change(self, filepath: str) -> bool:
+        main_script_path = self._session_data.main_script_path
+
+        page_for_file = next(
```

Comment:
I am surprised that there is no more elegant way to "find" element by custom predicat in Python.

Reply:
Yeah, this was similarly surprising to me, but I wasn't able to find another popular/pythonic method of doing this